provider/clusterrosa/hcp/resource.go

```diff
@@ -0,0 +1,1455 @@
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/proxy"
+
+	ocmr "github.com/terraform-redhat/terraform-provider-rhcs/internal/ocm/resource"
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/clusterrosa/hcp/upgrade"
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/clusterrosa/rosa"
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/clusterrosa/sts"
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/common"
+)
+
+const (
+	defaultTimeoutInMinutes   = int64(60)
+	nonPositiveTimeoutSummary = "Can't poll cluster state with a non-positive timeout"
+	nonPositiveTimeoutFormat  = "Can't poll state of cluster with identifier '%s', the timeout that was set is not a positive number"
+	pollingIntervalInMinutes  = 2
+
+	awsCloudProvider      = "aws"
+	rosaProduct           = "rosa"
+	MinVersion            = "4.10.0"
+	maxClusterNameLength  = 15
+	tagsPrefix            = "rosa_"
+	tagsOpenShiftVersion  = tagsPrefix + "openshift_version"
+	lowestHttpTokensVer   = "4.11.0"
+	propertyRosaTfVersion = tagsPrefix + "tf_version"
+	propertyRosaTfCommit  = tagsPrefix + "tf_commit"
+	waitTimeoutInMinutes  = 60
+)
+
+type ClusterRosaHcpResource struct {
+	clusterCollection *cmv1.ClustersClient
+	versionCollection *cmv1.VersionsClient
+	clusterWait       common.ClusterWait
+}
+
+var _ resource.ResourceWithConfigure = &ClusterRosaHcpResource{}
+var _ resource.ResourceWithImportState = &ClusterRosaHcpResource{}
+
+func New() resource.Resource {
+	return &ClusterRosaHcpResource{}
+}
+
+func (r *ClusterRosaHcpResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster_rosa_hcp"
+}
+
+func (r *ClusterRosaHcpResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "OpenShift managed cluster using rosa sts.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Unique identifier of the cluster.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					// This passes the state through to the plan, preventing
+					// "known after apply" since we know it won't change.
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"external_id": schema.StringAttribute{
+				Description: "Unique external identifier of the cluster. " + common.ValueCannotBeChangedStringDescription,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "Name of the cluster. Cannot exceed 15 characters in length. " + common.ValueCannotBeChangedStringDescription,
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtMost(15),
+				},
+			},
+			"cloud_region": schema.StringAttribute{
```

Comment:
Because [this](https://github.com/terraform-redhat/terraform-provider-rhcs/blob/02cc52087a080963975c18dc82e7041aa6d5cfc2/provider/clusterrosa/rosa/validators.go#L20) validator is shared, I'd say we should stick to `cloud_region` or we would need to duplicate the functionality

Reply:
Open a ticket for this one